pallets/sto/src/lib.rs

```diff
@@ -279,10 +279,11 @@ decl_module! {
                 Error::<T>::InvalidPriceTiers
             );
 
-            let offering_amount: T::Balance = tiers
-                .iter()
-                .map(|t| t.total)
-                .fold(0.into(), |total, x| total + x);
+            let mut offering_amount: T::Balance = 0.into();
+            for price_tier in tiers.iter() {
+                offering_amount = offering_amount.checked_add(&price_tier.total)
+                    .ok_or_else(|| Error::<T>::InvalidPriceTiers)?;
```

Comment:
Nit: I'd prefer keeping the functional style, which you can retain using `.try_fold(...)`.